src/CodeViewer/Viewer.cpp

```diff
@@ -72,15 +83,14 @@ Viewer::Viewer(QWidget* parent) : QPlainTextEdit(parent), left_sidebar_widget_{t
 void Viewer::resizeEvent(QResizeEvent* ev) {
   QPlainTextEdit::resizeEvent(ev);
 
-  auto left_sidebar = contentsRect();
-  left_sidebar.setWidth(left_sidebar_widget_.sizeHint().width());
-  left_sidebar_widget_.setGeometry(left_sidebar);
+  UpdateSidebarPositions();
```

Comment:
Why is it not also UpdateSidebarWidth?

Reply:
Maybe I am wrong, but because they only change when the font-size change (so, after zooming in-out and not a resize event)?

Reply:
That might be the case. Basically wanted to point you to this, as before it has the "left_sidebar.SetWidth".

Reply:
Yes, but we aren't changing the width there. If you read UpdateSidebarPositions we are also making that SetWidth.

This is code that was there before, but basically:
1 - Create a new left_sidebar with the entire windows size (  auto left_sidebar = contentsRect(); )
2- We set the width to the previous width (  left_sidebar.setWidth(left_sidebar_widget_.sizeHint().width()); )
3- We change our left_sidebar for the new one.

All these steps is to simply move the left_sidebar to the new position of the window without changing their size.

Reply:
`UpdateSidebarWidth` is only needed when something occured which leads to a different width of the sidebar. Since line numbers are shown there, one event that could cause a width-change is changing the shown code. A different number of lines might need more or less sidebar width due to more or fewer digits.

The same applies for changing the settings (enabling/disabling linenumbers/heatmap).

But in this event handler the only thing that actually changes is the size of the viewport, which means we might need to position the sidebar somewhere else, but we don't need to update its width.

Does that make sense?

Reply:
thanks